src/app/courses/add-course/add-course.component.ts

```diff
@@ -31,18 +32,28 @@ export class AddCourseComponent implements OnInit {
   }
 
   onSave(): void {
-    const course: Course = {
-      id: this.id,
-      title: this.title,
-      creationDate: this.date,
-      duration: this.duration,
+
+    let authorsList: HttpAuthor[] = [];
+
+    this.authors.forEach(author => {
+      authorsList.push({
+        id: Math.floor(1000 + Math.random() * 9000),
+        name: author
+      })
+    })
+
+    const course: HttpCourse = {
+      id: Math.floor(1000 + Math.random() * 9000),
+      name: this.title,
+      date: this.date.toString(),
+      length: this.duration,
       description: this.description,
-      topRated: false
+      isTopRated: false,
+      authors: authorsList
     }
 
     if (this.isEditMode) {
-      this.courseService.updateCourse(course);
```

Comment:
Does the _updateCourse_ should be deleted here?
Guess the patch request should be sent to modify the course

Reply:
There was no task for updating here, I guess it was left for Redux part.